packages/core/src/applicationcomposer/messageHandlers/generateResourceHandler.ts

```diff
@@ -72,13 +74,11 @@ async function generateResource(prompt: string) {
         let supplementaryWebLinks: SupplementaryWebLink[] = []
         let references: Reference[] = []
 
-        if (AuthUtil.instance.isConnectionExpired()) {
-            await AuthUtil.instance.showReauthenticatePrompt()
-        }
+        await amazonqApi.authApi.reauthIfNeeded()
```

Comment:
Under the hood this shows the prompt to reauth. If the prompt is shown, should this function (`generateResource()`) return early instead of continuing since it doesn't actually wait until the user is properly authenticated.

Reply:
It waits until the user is properly auth'd. If you open app composer, expire the connection, then hit generate suggestions it will prompt you to authenticate then it just waits until the user is authenticated before continuing